src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -176,22 +177,32 @@ public WebSocketsTransport(WebSocketOptions options, Channel<byte[]> application
                     {
                         try
                         {
-                            _logger.SendPayload(_connectionId, buffer.Length);
+                            _logger.SendPayload(_connection.ConnectionId, buffer.Length);
+
+                            var webSocketMessageType = _options.WebSocketMessageType;
+                            var transferModeFeature = _connection.Features.Get<ITransferModeFeature>();
+                            if (transferModeFeature != null)
```

Comment:
It actually change dynamically so can't be passed to the ctor since we don't know. We have a negotiation message where the client selects the protocol which tells us what WebSocketMessageType to use.